### PR TITLE
LocalStorage + Calendar 

### LINKED ISSUE #10: Add localStorage
## Value statement ##
As a **user**
I need to **be able to save my habits**
so **I can retrieve and check them every day** 

## Description ##
Bildschirmfoto 2020-08-29 um 19 42 40

## Acceptance criteria ##

- [x] Checkboxes in habits can be saved and retrieved
- [x] Habits/Categories can be saved
- [x] Habits/Categories can be retrieved every time user opens the app

## Task ##

- [x] Set up localStorage
- [x] Save added habits/categories automatically on add
- [x] Retrieve saved habits/categories in the dashboard, with their dates matching

## Complexity ##
small

### PR DESCRIPTION
**Pull request by denizulku**
Fixes #10 
References #3 
Please review & keep in mind that my libraries for the color picker and the calender **do not** support StyledComponents. Which is why I had to either do inline styling or styling through the libraries' own pre-defined constants.

### CODE REVIEW

File: src/GlobalStyles.js [hunk 1]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
...
    --color-fill: white;
    --color-create-icon: white;
-   --color-cancel-icon: #ffb9b9;
+   --color-cancel-icon: #f99;
    --color-nav-bg: #bcbbbc;
+   --color-calendar-sundays: red;
  }
+
+ p {
+    font-family: Helvetica;
```
Comment (soerenstein) [inline src/GlobalStyles.js:41]: Are there more fonts in your app? If not, you can also use `@font-face`.

File: src/App.js [hunk 2]
```diff
@@ -12,10 +12,10 @@ function App() {
 
   return (
     <Router>
-      <Header />
       <main>
         <Switch>
           <Route exact path="/">
+            <Header />
```
Comment (maximiliangustmann) [inline src/App.js]: Why did you move your header inside your switch, if it is on every page?

File: src/GlobalStyles.js [hunk 3]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
 
 export default createGlobalStyle`
 body {
-   font-family: sans-serif;
+   box-sizing: border-box;
```
Comment (maximiliangustmann) [inline src/GlobalStyles.js:6]: I would move the box-sizing to the universal selector *

File: src/GlobalStyles.js [hunk 4]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
...
+   margin: 20px;
```
Comment (maximiliangustmann) [inline src/GlobalStyles.js:11]: Is this a thing? I thought it would be common to give the body a margin of 0.

**Review by soerenstein — COMMENTED** [review]
Good job! 🚀👍🏼 I just added a comment.

File: src/GlobalStyles.js [hunk 5]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
...
 }
 
-h2 {
-   margin-top: 10px;
-   font-size: 16px;
-   position: relative;
+h1 {
+   font-family: Helvetica;
```
Comment (maximiliangustmann) [inline src/GlobalStyles.js]: Do you use other font-familys as well? Otherwise I would move this to the universal selector *.

File: src/GlobalStyles.js [hunk 6]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
...
+   font-size: 18px;
```
Comment (maximiliangustmann) [inline src/GlobalStyles.js]: A could use a more responsive size like em or rem

File: src/GlobalStyles.js [hunk 7]
```diff
@@ -2,20 +2,28 @@ import { createGlobalStyle } from 'styled-components/macro'
...
+   margin-bottom: 30px;
 }
 
 html {
    --color-fill: white;
    --color-create-icon: white;
-   --color-cancel-icon: #ffb9b9;
+   --color-cancel-icon: #f99;
    --color-nav-bg: #bcbbbc;
+   --color-calendar-sundays: red;
```
Comment (maximiliangustmann) [inline src/GlobalStyles.js]: Would move this into :root

File: src/components/Header/Header.js [hunk 8]
```diff
@@ -1,29 +1,18 @@
 import React from 'react'
 import styled from 'styled-components/macro'
-import HeaderLogo from '../../components/assets/Logo/HeaderLogo.svg'
+import { ReactComponent as HeaderLogo } from '../../components/assets/Logo/HeaderLogo.svg'
 import PropTypes from 'prop-types'
```
Comment (maximiliangustmann) [inline src/components/Header/Header.js]: Do you use PropTypes here?

**Review by maximiliangustmann — COMMENTED** [review]
Good job!
